smack-extensions/src/main/java/org/jivesoftware/smackx/usertune/element/UserTuneElement.java

```diff
@@ -0,0 +1,258 @@
+        if (this.artist == null && this.length == null && this.rating == null && this.source == null && this.title == null
+                && this.track == null && this.uri == null) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static boolean hasUserTuneElement(Message message) {
+        return message.hasExtension(ELEMENT, NAMESPACE);
+    }
+
+    public static UserTuneElement fromMessage(Message message) {
+        return message.getExtension(ELEMENT, NAMESPACE);
+    }
+
+    public UserTuneElement getUserTune() {
+        return this;
+    }
+
+    /**
+     * Returns a new instance of {@link Builder}.
+     * @return Builder
+     */
+    public static Builder getBuilder() {
+        return new Builder();
+    }
+
+    /**
+     * This class defines a Builder class for {@link UserTuneElement}.
+     * <br>
+     * {@link UserTuneElement} instance can be obtained using the {@link #build()} method as follows.
+     * <br>
+     * UserTuneElement.Builder builder = new UserTuneElement.Builder();
+     * builder.setSource("Yessongs"); builder.setTitle("Heart of the Sunrise");
+     * UserTuneElement userTuneElement = builder.build();
+     * <br>
+     * Values such as title, source, artist, length, source, track and uri can be set using their respective setters through {@link Builder}
+     * <br>
+     * Use {@link #buildEmptyUserTune()} only while disabling UserTune publish through {@link UserTuneManager#clearUserTune()}
+     */
+    public static final class Builder {
+        private String artist;
+        private Integer length;
+        private Integer rating;
+        private String source;
+        private String title;
+        private Integer track;
+        private URI uri;
+
+        private Builder() {
+        }
+
+        /**
+         * Artist is an optional element in UserTuneElement.
+         * @param artist
+         * @return Builder
+         */
+        public Builder setArtist(String artist) {
+            this.artist = artist;
+            return this;
+        }
+
+        /**
+         * Length is an optional element in UserTuneElement.
+         * @param length
+         * @return Builder
+         */
+        public Builder setLength(int length) {
+            if (length > 0) {
+                this.length = length;
+                return this;
+            }
+            else {
+                throw new IllegalArgumentException("Length should be positive. In case of live streaming, length is expected to be null");
+            }
+        }
+
+        /**
+         * Rating is an optional element in UserTuneElement.
+         * The rating to be given is suppossed to be out of 10.
+         * <br>
+         * Method will throw an Exception if rating is less than 0 or greater than 10.
+         * @param rating
+         * @return Builder
+         */
+        public Builder setRating(int rating) {
+            if (rating <= 10 && rating >= 0) {
```

Comment:
This allows for rating to be 0, which is invalid as far as I can tell from XEP-0118.

Also should be simplified to:
```
if (rating < 1 || rating > 10) {
    throw new IllegalArgumentException("Meaningful exception message");
}
this.rating = rating;
return this;
````